ZenPacks/zenoss/CalculatedPerformance/dsplugins.py

```diff
@@ -77,6 +81,22 @@ def handleArguments(targetArgValues, dpargs):
     return arguments
 
 
+def getCollectionInterval(cachedValue, minimumInterval, maximumInterval):
+    """Determine collection interval based on delta 
+    between the last and previous data collection.
+
+    minimumInterval and maximumInterval attributes limit the interval.
+    """
+    previousCollectionTime = cachedValue[0][1]
+    lastCollectionTime = cachedValue[1][1]
+    interval = lastCollectionTime - previousCollectionTime
+    if interval < minimumInterval:
```

Comment:
From here on down we'd probably want to do something like this to enforce a hard minimum of 1, and the case that minimumInterval or maximumInterval is None.

```python
minimum = interval if minimumInterval is None else minimumInterval
maximum = interval if maximumInterval is None else maximumInterval
return max(1, max(min(maximum, interval), minimum))
```